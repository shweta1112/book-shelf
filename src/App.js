import React, { useState, useEffect } from "react";
import * as BooksAPI from "./serviceCalls/BooksAPI";
import BookInput from "./components/book_input/BookInput.component";
import MyBookShelf from "./components/my_book_shelf/MyBookShelf.component";
import "./App.css";
import { Route } from "react-router-dom";

const BooksApp = () => {
  const [books, setBooks] = useState({});
  const [wantToReadBooks, setWantToReadBooks] = useState([]);
  const [currentlyReadingBooks, setCurrentlyReadingBooks] = useState([]);
  const [readBooks, setReadBooks] = useState([]);

  useEffect(() => {
    const getBooks = () => {
      BooksAPI.getAll().then((booksArray) => {
        booksArray.map((book) => (books[book.id] = book));
        setBooks(books);
        setWantToReadBooks(
          booksArray
            .filter((book) => book.shelf === "wantToRead")
            .map((book) => book.id)
        );
        setCurrentlyReadingBooks(
          booksArray
            .filter((book) => book.shelf === "currentlyReading")
            .map((book) => book.id)
        );
        setReadBooks(
          booksArray
            .filter((book) => book.shelf === "read")
            .map((book) => book.id)
        );
      });
    };
    getBooks();
  }, [books]);

  const getBookUpdated = (bookid, event) => {
    const selectedValue = event.target.value;
    BooksAPI.get(bookid)
      .then((book) => {
        books[bookid] = book;
      })
      .then(() => {
        books[bookid].shelf = selectedValue;
        BooksAPI.update(bookid, selectedValue).then((resp) => {
          setCurrentlyReadingBooks(resp["currentlyReading"]);
          setWantToReadBooks(resp["wantToRead"]);
          setReadBooks(resp["read"]);
        });
      });
  };

  return (
    <div className="app">
      <Route
        exact
        path="/"
        render={() => (
          <MyBookShelf
            key={books.title}
            books={books}
            getBookUpdated={getBookUpdated}
            currentlyReadingBooks={currentlyReadingBooks}
            wantToReadBooks={wantToReadBooks}
            readBooks={readBooks}
          />
        )}
      />
      <Route
        path="/search"
        render={({ history }) => (
          <BookInput
            books={books}
            getBookUpdated={(book, shelf) => {
              getBookUpdated(book, shelf);
              history.push("/");
            }}
          />
        )}
      />
    </div>
  );
};

export default BooksApp;
