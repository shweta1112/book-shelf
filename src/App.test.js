import React from "react";
import { configure, mount } from "enzyme";
import { BrowserRouter } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import App from "./App";

/**
 This course is not designed to teach Test Driven Development.
 Feel free to use this file to test your application, but it
 is not required.
**/
configure({ adapter: new Adapter() });
const wrapper = () =>
  mount(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
describe("<App/>", () => {
  // it("renders without crashing", () => {
  //   const div = document.createElement("div");
  //   ReactDOM.render(
  //     <BrowserRouter>
  //       <App />
  //     </BrowserRouter>,
  //     div
  //   );
  // });
  it("should match snapshort", () => {
    const snapshort = renderer.create(wrapper).toJSON();
    expect(snapshort).toMatchSnapshot();
  });
});
