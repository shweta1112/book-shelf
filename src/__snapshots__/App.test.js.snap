// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should match snapshort 1`] = `null`;
